base/essentials.jl

```diff
@@ -234,10 +234,11 @@ function unwrap_unionall(@nospecialize(a))
 end
 
 function rewrap_unionall(@nospecialize(t), @nospecialize(u))
-    if !isa(u, UnionAll)
+    if isa(u, UnionAll)
+        return UnionAll(u.var, rewrap_unionall(t, u.body))
+    else
         return t
     end
```

Comment:
This had to change as `!` is not defined in the world-age I wanted to call it from.
But I think the new version is just as readable